marketplace-service/src/main/java/com/axonivy/market/util/VersionUtils.java

```diff
@@ -142,7 +141,7 @@ public static List<String> getInstallableVersionsFromMetadataList(List<Metadata>
       return new ArrayList<>();
     }
     return metadataList.stream().filter(MavenUtils::isProductMetadata).findAny().map(
-        metadata -> metadata.getVersions().stream().sorted(new LatestVersionComparator()).collect(
-            Collectors.toList())).orElse(new ArrayList<>());
+        metadata -> metadata.getVersions().stream().sorted(new LatestVersionComparator()).toList()).orElse(
```

Comment:
did you check this method getVersionsForDesigner() ?
when to use toList, it will create an immutable collection that we can not filter and .toList again.